docker/docker_client.go

```diff
@@ -412,7 +415,13 @@ func getAuth(ctx *types.SystemContext, registry string) (string, string, error)
 	return "", "", nil
 }
 
-func (c *dockerClient) ping() error {
+// ensurePropertiesAreDetected detects various properties of the registry.
+// See the dockerClient documentation for members which are affected by this.
+func (c *dockerClient) ensurePropertiesAreDetected() error {
```

Comment:
ACK, `detectProperties` works just as well (and the callers don’t care that we cache the data, no need to expose that in the name.)